internal/coq/coq.go

```diff
@@ -759,6 +773,35 @@ func (e SpawnExpr) Coq() string {
 	return pp.Build()
 }
 
+// FuncLit is an unnamed function literal, consisting of its parameters and body.
+type FuncLit struct {
+	Args []FieldDecl
+	// TODO: ReturnType Type
+	Body Expr
+	// TODO: AddTypes   bool
+}
+
+func (e FuncLit) Coq() string {
+	var pp buffer
+
+	var args []string
+	for _, a := range e.Args {
+		args = append(args, a.CoqBinder())
+	}
+	if len(args) == 0 {
+		args = []string{"<>"}
+	}
+	sig := strings.Join(args, " ")
+
+	pp.Add("(rec: \"\" %s :=", sig)
```

Comment:
You can just use a lambda construct here (`λ: %s,`). This is actually a recursive function whose name is the empty string (which is a valid variable name).